runtime/appruntime/metrics/metrics.go

```diff
@@ -0,0 +1,47 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+
+	"encore.dev/beta/errs"
+)
+
+type Manager struct {
+	exp Exporter
+}
+
+type Exporter interface {
+	IncCounter(name string, tags ...string)
+	Observe(name string, key string, value float64, tags ...string)
+}
+
+func NewManager(exp Exporter) *Manager {
+	return &Manager{exp: exp}
+}
+
+func (m *Manager) ReqBegin(service, endpoint string) {
+	m.exp.IncCounter("e_requests_total", "service", service, "endpoint", endpoint)
+}
+
+func (m *Manager) ReqEnd(service, endpoint string, err error, httpStatus int, durMillis int64) {
+	if err != nil {
+		e := errs.Convert(err).(*errs.Error)
+		m.exp.IncCounter("e_errors_total", "service", service, "endpoint", endpoint, "code", e.Code.String())
+
+		if httpStatus == 0 {
+			httpStatus = e.Code.HTTPStatus()
+		}
+	}
+
+	if httpStatus == 0 {
+		httpStatus = http.StatusOK
+	}
+	m.exp.Observe(
+		"e_request_durations_milliseconds",
+		"duration", float64(durMillis),
+		"service", service,
+		"endpoint", endpoint,
+		"status_code", strconv.Itoa(httpStatus),
```

Comment:
`status_code` differs from `code` (used above) in that it's not supposed to be human-readable. This is why they're not named the same.

For consistency, we could name them both `code` and convert the HTTP status code into a string such as `http_201`, for example. Alternatively, we could use the `StatusText` function (see [docs](https://pkg.go.dev/net/http#StatusText)) to convert an HTTP status code into a human-readable code.